WcaOnRails/spec/controllers/competitions_controller_spec.rb

```diff
@@ -644,383 +644,61 @@
-
-            get :post_results, params: { id: competition, event_id: "333mbf" }
-            post = assigns(:post)
-            expect(post.title).to eq "Jeremy wins #{competition.name}, in #{competition.cityName}, #{competition.country.name_in(:en)}"
-            expect(post.body).to eq "[Jeremy](#{person_url('2006YOYO01')}) won [#{competition.name}](#{competition_url(competition)}) with a result of 8/9 45:32 in the 3x3x3 Multi-Blind event. " \
-              "[Dan](#{person_url('2006YOYO02')}) finished second (8/9 45:32) and " \
-              "[Steven](#{person_url('2006YOYO03')}) finished third (8/9 45:32).\n\n"
-          end
+      it "sends the notification emails to users that competed" do
+        FactoryBot.create_list(:user_with_wca_id, 4, results_notifications_enabled: true).each do |user|
+          FactoryBot.create_list(:result, 2, person: user.person, competitionId: competition.id, eventId: "333")
         end
-      end
 
-      it "announces world records" do
-        Result.create!(
-          pos: 1,
-          personId: "2006SHEU01",
-          personName: "Vincent Sheu",
-          countryId: "USA",
-          competitionId: competition.id,
-          eventId: "333fm",
-          roundTypeId: "f",
-          formatId: "m",
-          value1: 25,
-          value2: 26,
-          value3: 27,
-          best: 25,
-          average: 2600,
-          regionalSingleRecord: "WR",
-          regionalAverageRecord: "WR",
-        )
-        # Another Vincent Sheu!
-        Result.create!(
-          pos: 1,
-          personId: "2006SHEU02",
-          personName: "Vincent Sheu",
-          countryId: "USA",
-          competitionId: competition.id,
-          eventId: "222",
-          roundTypeId: "f",
-          formatId: "m",
-          value1: 1000,
-          value2: 2000,
-          value3: 3000,
-          best: 1000,
-          average: 2000,
-          regionalSingleRecord: "WR",
-          regionalAverageRecord: "",
-        )
-        Result.create!(
-          pos: 1,
-          personId: "2005FLEI01",
-          personName: "Jeremy Fleischman",
-          countryId: "USA",
-          competitionId: competition.id,
-          eventId: "333oh",
-          roundTypeId: "f",
-          formatId: "m",
-          value1: 4000,
-          value2: 5000,
-          value3: 6000,
-          best: 4000,
-          average: 5000,
-          regionalSingleRecord: "NAR",
-          regionalAverageRecord: "WR",
-        )
-        Result.create!(
-          pos: 1,
-          personId: "2005FLEI01",
-          personName: "Jeremy Fleischman",
-          countryId: "USA",
-          competitionId: competition.id,
-          eventId: "333oh",
-          roundTypeId: "1",
-          formatId: "m",
-          value1: 4100,
-          value2: 5100,
-          value3: 6100,
-          best: 4100,
-          average: 5100,
-          regionalSingleRecord: "NAR",
-          regionalAverageRecord: "",
-        )
```

Comment:
All of those removed tests could actually test the results helper that handles this logic now.